_source/_posts/2022-09-12-react-native-spring-boot-jhipster.adoc

```diff
@@ -0,0 +1,531 @@
+
+[cols="a,a", frame=none, grid=none]
+|===
+| image::{% asset_path blog/react-native-jhipster/react-native-web.jpg %}[alt=React Native web,width=400,align=center,link={% asset_path blog/react-native-jhipster/react-native-web.jpg %}]
+| image::{% asset_path blog/react-native-jhipster/react-native-web-photos.jpg %}[alt=React Native photos list,width=400,align=center,link={% asset_path blog/react-native-jhipster/react-native-web-photos.jpg %}]
+|===
+
+[cols="a,a", frame=none, grid=none]
+|===
+| image::{% asset_path blog/react-native-jhipster/react-native-web-photo.jpg %}[alt=React Native photo with Josh,width=400,align=center,link={% asset_path blog/react-native-jhipster/react-native-web-photo.jpg %}]
+| image::{% asset_path blog/react-native-jhipster/react-native-edit-photo.jpg %}[alt=Edit photo,width=400,align=center,link={% asset_path blog/react-native-jhipster/react-native-edit-photo.jpg %}]
+|===
+
+=== Test on iOS
+
+To see your React Native app running on iOS, press **i** in the window you ran `npm start` from. You will need to be on a Mac with Xcode installed for this to work.
+
+[cols="a,a", frame=none, grid=none]
+|===
+| image::{% asset_path blog/react-native-jhipster/react-native-ios.png %}[alt=React Native app on iOS,width=350,align=center]
+| image::{% asset_path blog/react-native-jhipster/react-native-ios-photo.jpg %}[alt=React Native photo on iOS,width=350,align=center]
+|===
+
+TIP: You can reload your app in Simulator using kbd:[{commandKey} + R].
+
+=== Test on Android
+
+To see your React Native app running on Android, press **a** in the window you ran `npm start` from. You will need Android Studio and an AVD (Android Virtual Device) running. I tested on a Pixel 5 with API 31 (Android 12.0).
+
+For the Android emulator to communicate with your API and Keycloak, you'll need to add some port mappings. The command worked if `8080` and `9080` are printed to your terminal.
+
+[source,shell]
+----
+adb reverse tcp:8080 tcp:8080 && adb reverse tcp:9080 tcp:9080
+----
+
+CAUTION: If you get a `command not found` error, see https://stackoverflow.com/questions/10303639/adb-command-not-found[this Stack Overflow Q & A] to solve it. I used `echo export "PATH=~/Library/Android/sdk/platform-tools:$PATH" >> ~/.zshrc` on my Mac.
+
+To reload your app, hit kbd:[r] twice with a focus on the Android emulator.
+
+[cols="a,a,a", frame=none, grid=none]
+|===
+| image::{% asset_path blog/react-native-jhipster/android-expo-prompt.png %}[alt=Android Expo prompt,width=300,align=center]
+| image::{% asset_path blog/react-native-jhipster/android-keycloak-login.png %}[alt=Android Keycloak login,width=300,align=center]
+| image::{% asset_path blog/react-native-jhipster/android-keycloak-signed-in.png %}[alt=Android Keycloak login,width=300,align=center]
+|===
+
+== Use Auth0 for Identity
```

Comment:
```suggestion
== Use Auth0 for identity
```

Reply:
I actually did this one on purpose because I liked the look of Identity in uppercase. I'll commit your suggestion through.